### PR TITLE
Add a section on writing tool templates to the chat template docs

### PR DESCRIPTION
**Pull request by Rocketknight1**
The chat template docs have an extensive section on using tool templates, but they're missing good guidance to template authors. I added a section to help them, and also so there's a point of reference for the API we expect tool templates to follow.

### CODE REVIEW

File: docs/source/en/chat_templating.md [hunk 1]
```diff
@@ -962,4 +962,128 @@ tokenizer.chat_template = open("template.jinja").read()
...
+          "description": "The second number to multiply"
+        }
+      }, 
+      "required": ["a", "b"]
+    }
+  }
+}
+```
+
+A common pattern for handling tools in chat templates would be something like this:
```
Comment (pcuenca) [inline docs/source/en/chat_templating.md:1009]: It is clear that these are examples and there's also a mention at the end, but maybe we can reinforce even more and explicitly state it here.

**Review by pcuenca — APPROVED** [review]
Super clear, I really like it! I hope model authors read it 😅

File: docs/source/en/chat_templating.md (hunk 1 above)
Comment (Rocketknight1) [reply docs/source/en/chat_templating.md]: Done!

**Comment by Rocketknight1** [comment]
I'm happy with this now - merging without core maintainer review since it's a pure documentation change

**Pull request merged**